corehq/apps/registration/models.py

```diff
@@ -170,6 +170,7 @@ class SelfSignupWorkflow(models.Model):
     )
 
     @classmethod
+    @memoized
```

Comment:
I think you want quickcache for this. Memoized only lasts for the duration of the request. You could probably quickcache this value for quite a while.